test/e2e/data/capi-operator/azure-provider.yaml

```
@@ -6,8 +6,3 @@ metadata:
   namespace: capz-system
 spec:
   type: infrastructure
-  deployment:
-    containers:
-      - name: manager
-        args:
```

This is overriding the args, removing all the defaults, for example MachinePool feature support.
It would be best to fix this upstream, allowing for debug level configuration using an environment variable.

For now I think it's best to just remove it, I find the normal log level enough actually.

---

for CAPZ specifically, the default log level is not enough. Sometimes it hides errors related to for example credentials.

---

Then again I'd rather fix this upstream, I don't think overriding the args is supported and should be used anyway, as we will diverge from upstream.